src/Examine.Test/Examine.Lucene/Search/FluentApiTests.cs

```diff
@@ -2556,6 +2556,141 @@ public void Given_SkipTake_Returns_ExpectedTotals(int skip, int take, int expect
+
+                // Third query result continues after result 2 (zero indexed), Takes 1
+                var searchAfter2 = new SearchAfterOptions(luceneResults2.SearchAfter.DocumentId, luceneResults2.SearchAfter.DocumentScore, luceneResults2.SearchAfter.Fields, luceneResults2.SearchAfter.ShardIndex.Value);
+                var luceneOptions3 = new LuceneQueryOptions(0, 1, searchAfter2);
+                var results3 = sc.Execute(luceneOptions3);
+                var luceneResults3 = results3 as ILuceneSearchResults;
+                Assert.IsNotNull(luceneResults3);
+                var luceneResults3List = luceneResults3.ToList();
+                Assert.IsTrue(luceneResults3List.Any(x => x.Id == "4"), $"Expected to contain next result after docId {luceneResults2.SearchAfter.DocumentId}");
+                Assert.IsFalse(luceneResults3.ToList().Any(x => luceneResults2.Any(y => y.Id == x.Id)), "Results should not overlap");
+                Assert.IsFalse(luceneResults3.ToList().Any(x => luceneResults.Any(y => y.Id == x.Id)), "Results should not overlap");
+
+                Assert.AreNotEqual(results.First().Id, results2.First().Id, "Results should be different");
+            }
+        }
+
+        [Test]
+        public void SearchAfter_NonSorted_Results_Returns_Different_Results()
+        {
+            var analyzer = new StandardAnalyzer(LuceneInfo.CurrentVersion);
+            using (var luceneDir = new RandomIdRAMDirectory())
+            using (var indexer = GetTestIndex(luceneDir, analyzer))
+            {
+                indexer.IndexItems(new[] {
+                    ValueSet.FromObject(1.ToString(), "content",
+                        new { nodeName = "umbraco", headerText = "world", writerName = "administrator" }),
+                    ValueSet.FromObject(2.ToString(), "content",
+                        new { nodeName = "umbraco", headerText = "umbraco", writerName = "administrator" }),
+                    ValueSet.FromObject(3.ToString(), "content",
+                        new { nodeName = "umbraco", headerText = "umbraco", writerName = "administrator" }),
+                    ValueSet.FromObject(4.ToString(), "content",
+                        new { nodeName = "umbraco", headerText = "nz", writerName = "administrator" }),
+                    ValueSet.FromObject(5.ToString(), "content",
+                        new { nodeName = "hello", headerText = "world", writerName = "blah" })
+                    });
+
+                var searcher = indexer.Searcher;
+
+                //Arrange
+                var sc = searcher.CreateQuery("content")
+                    .Field("writerName", "administrator");
+                var luceneOptions = new LuceneQueryOptions(0, 2);
+                //Act
+
+                //There are 4 results
+                // First query skips 0 and takes 2.
+                var results = sc.Execute(luceneOptions);
```

Comment:
Wonder if we can add some syntactic sugar here with an extension method like ExecuteWithLucene (or similar) which does all the type checking, throws any NotSupportedExceptions (if it's not Lucene based) and returns ILuceneSearchResults. 

Alternatively, could have the extension method still called Execute but it will be Execute{T} where T : ISearchResults and do the type checking, validation, etc... in there.

Reply:
Added ExecuteWithLucene